components/integrations_marketplace/marketplace_item/marketplace_item.jsx

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {FormattedMessage} from 'react-intl';
+import {Link} from 'react-router-dom';
+
+import {MarketplacePluginStatus} from 'mattermost-redux/constants/plugins';
+
+import LoadingWrapper from 'components/widgets/loading/loading_wrapper.jsx';
+import ConfirmModal from 'components/confirm_modal.jsx';
+import PluginIcon from 'components/widgets/icons/plugin_icon.jsx';
+
+import {localizeMessage} from 'utils/utils';
+
+export default class MarketplaceItem extends React.Component {
+        static propTypes = {
+            id: PropTypes.string.isRequired,
+            name: PropTypes.string.isRequired,
+            description: PropTypes.string.isRequired,
+            version: PropTypes.string.isRequired,
+            downloadUrl: PropTypes.string,
+            homepageUrl: PropTypes.string,
+            installed: PropTypes.bool.isRequired,
+            onConfigure: PropTypes.func.isRequired,
+            onInstalled: PropTypes.func.isRequired,
+            actions: PropTypes.shape({
+                installPluginFromUrl: PropTypes.func.isRequired,
+            }).isRequired,
+        };
+
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                installed: this.props.installed,
+                installing: false,
+                confirmOverwriteInstallModal: false,
+                serverError: null,
+            };
+        }
+
+        onClickOvewriteModal = () => {
```

Comment:
Do we still need a confirmation flow for overwriting (aka upgrading, I presume)?